local_planner/src/nodes/common.h

```diff
@@ -10,58 +10,53 @@
+  float e;
+  float z;
+  float r;
+};
+
+float distance2DPolar(const PolarPoint& p1, const PolarPoint& p2);
 
 /**
 * @brief     Convertes the point from polar CS to cartesian CS
-* @param[in] e elevation angle (-90, 90]
-* @param[in] z azimuth angle (-180,180]
-* @param[in] radius
+* @param[in] p_pol polar point
 * @param[in] pos Position from which to convert the point
 * @returns   point in cartesian CS
 **/
-Eigen::Vector3f fromPolarToCartesian(float e, float z, double radius,
-                                     const geometry_msgs::Point& pos);
+Eigen::Vector3f PolarToCartesian(const PolarPoint& p_pol,
+                                 const geometry_msgs::Point& pos);
 double indexAngleDifference(float a, float b);
 
-double elevationIndexToAngle(int e, double res);
-double azimuthIndexToAngle(int z, double res);
+PolarPoint HistogramIndexToPolar(int e, int z, int res, const float& radius);
 
 /**
-* @brief     Compute the bearing angle of a given vector relative to an
+* @brief     Compute the polar point to an
 *            origin.
 * @param[in] position Position of the location to which to compute the bearing
 *            angle to.
 * @param[in] origin Origin from which to compute the bearing vector.
 * @details   For a point given in cartesian x/y coordinates this is the
 *            angle in degrees from the positive y-axis in (-180, 180].
 *
-* @returns   Angle in integer degrees from the positive y-axis (-180, 180]
+* @returns   azimuth Angle in float degrees from the positive y-axis (-180, 180]
+*and elevation angle degrees (-90, 90]
 * @warning   If the origin and the position coincide, the output is 0 degrees
 **/
-float azimuthAnglefromCartesian(const Eigen::Vector3f& position,
-                                const Eigen::Vector3f& origin);
-float azimuthAnglefromCartesian(double x, double y, const Eigen::Vector3f& pos);
-/**
-* @brief   Compute the elevation angle for a point given in cartesian
-*coordinates
-* @note    Output is in degrees (-90, 90)
-**/
-float elevationAnglefromCartesian(const Eigen::Vector3f& pos,
-                                  const Eigen::Vector3f& origin);
-float elevationAnglefromCartesian(double x, double y, double z,
-                                  const Eigen::Vector3f& pos);
+
+PolarPoint CartesianToPolar(const Eigen::Vector3f& pos,
+                            const Eigen::Vector3f& origin);
+PolarPoint CartesianToPolar(double x, double y, double z,
+                            const Eigen::Vector3f& pos);
 /**
-* @brief     Checks if the computed histogram index given an elevation angle and
-*resolution is valid
-* @param[in] e Elevation angle in degrees
+* @brief     compute polar point to histogram index
+* @param[in] p_pol with elevation, azimuth angle and radius
 * @param[in] res resolution of the histogram in degrees
-* @param[in] index Index of the histogram cell for the given elevation
-* @returns   true or
+* @param[out] eigenvector with x()=azimuth and y()=elevation
```

Comment:
eigenvector is a bit loaded. I would just write vector, the format is specified in the function